### PR TITLE
Fix incorrect assertions in BasicWorkflow

### PR DESCRIPTION
**Pull request by VC1995**
#1793 
Removed `asyncHelper.fishForMessage` and tried to determine `NewScalaNotesEvent` at different points.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ pipeline:
           exit 1 ;
         fi
       - if [ -n "$DRONE_PULL_REQUEST" ] ; then cd testing/cache && sbt ++$SCALA_VERSION ensimeConfig ensimeServerIndex && cd ../.. ; fi
-      - if [ -n "$DRONE_PULL_REQUEST" ] && [ "$SCALA_VERSION" = "2.12.2" ] ; then sbt ++$SCALA_VERSION "core/it:testOnly *BasicWorkflow*" ; fi
+      - if [ -n "$DRONE_PULL_REQUEST" ] ; then sbt ++$SCALA_VERSION "core/it:testOnly *BasicWorkflow*" ; fi
```
Comment (fommil) [inline .drone.yml]: to control the version of scala, remove entries from the matrix at the bottom of the file.

Comment (VC1995) [inline .drone.yml]: okay, but I was only testing a theory. No longer needed. :)

**Comment by VC1995** [comment]
So it happened because we get a `FullTypeCheckCompleteEvent` when we restart the compiler with non-empty files and the analyzer is in "ready" state (here https://github.com/ensime/ensime-server/blob/2.0/core/src/main/scala/org/ensime/core/Analyzer.scala#L134 and here https://github.com/ensime/ensime-server/blob/2.0/core/src/main/scala/org/ensime/core/Analyzer.scala#L162)
we were only receiving `CompilerRestartedEvent` in `BasicWorkflow` and the unaccounted message kept creeping down unless we saw some unpredicted message later on.

**Comment by fommil** [comment]
@VC1995 you've still got the temporary comments in the CI definition files.

File: core/src/it/scala/org/ensime/fixture/ProjectFixture.scala [hunk 2]
```diff
@@ -3,12 +3,11 @@
 package org.ensime.fixture
 
 import akka.actor.{ ActorRef, ActorSystem }
-import org.scalatest._
 import akka.testkit._
-
 import org.ensime.api._
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/ProjectFixture.scala]: could you please back out the import rearrangements?

Probably you need to change your IntelliJ sorting options so they go

```
java.

scala.

<other>
```

I'm not sure how you do that

File: core/src/it/scala/org/ensime/intg/BasicWorkflow.scala [hunk 3]
```diff
@@ -2,18 +2,18 @@
 // License: http://www.gnu.org/licenses/gpl-3.0.en.html
 package org.ensime.intg
 
-import scala.concurrent.duration._
```
Comment (fommil) [inline core/src/it/scala/org/ensime/intg/BasicWorkflow.scala]: same here

File: core/src/it/scala/org/ensime/intg/BasicWorkflow.scala [hunk 4]
```diff
@@ -37,24 +37,25 @@ class BasicWorkflow extends EnsimeSpec
 
           project ! TypecheckModule(EnsimeProjectId("testing_simple", "compile"))
           expectMsg(VoidResponse)
-          all(asyncHelper.receiveN(2)) should matchPattern {
+          all(asyncHelper.receiveN(3)) should matchPattern {
```
Comment (fommil) [inline core/src/it/scala/org/ensime/intg/BasicWorkflow.scala]: heh, nice catch. I actually tried something like this lower down, but it was after the fishing so it was missed. We should probably rationalise our use of `fish` everywhere.

File: core/src/it/scala/org/ensime/fixture/ProjectFixture.scala (hunk 2 above)
Comment (VC1995) [inline core/src/it/scala/org/ensime/fixture/ProjectFixture.scala]: I'm sorry that I keep messing it up. I'll fix this. :)

Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/ProjectFixture.scala]: heh, don't worry... everybody does.

**Comment by fommil** [comment]
you forgot to compile it in `sbt` before pushing so the formatting rules didn't kick in :smile:

**Pull request merged**